src/libcrun/cgroup-resources.c

```diff
@@ -244,8 +244,8 @@ write_blkio_v1_resources_throttling (int dirfd, const char *name, throttling_s *
     {
       int ret;
       size_t len;
-      len = sprintf (fmt_buf, "%" PRIu64 ":%" PRIu64 " %" PRIu64 "\n", throttling[i]->major, throttling[i]->minor,
-                     throttling[i]->rate);
+      len = snprintf (fmt_buf, sizeof (fmt_buf), "%" PRIu64 ":%" PRIu64 " %" PRIu64 "\n", throttling[i]->major, throttling[i]->minor,
```

Comment:
Should this be `sizeof(fmt_buf) -1`

Reply:
thanks for noticing it.

I've added a check as the remaining `snprintf` to fail if we used the entire buffer and have no room for '\0'